jsapp/js/dataInterface.ts

```diff
@@ -173,6 +177,13 @@ export interface SubmissionSupplementalDetails {
   }
 }
 
+/**
+ * This is a completely empty object.
+ *
+ * We can't use `{}`, as it means "any non-nullish value". We are using `Record<string, never>` as the closes thing.
```

Comment:
I don't understand why `{} | undefined` is not sufficient given the comment on line 246 below

Reply:
If I use `{}` then in the mocks the objects with missing propertes are not being caught :(